include/exadg/operators/generalized_laplace_operator.h

```diff
@@ -0,0 +1,849 @@
+
+    constant_coefficient = coefficient;
+  }
+
+  /**
+   * Recomputes `tau` before a face evaluation.
+   */
+  void
+  reinit_face(IntegratorFace &   integrator_m,
+              IntegratorFace &   integrator_p,
+              unsigned int const dof_index) const
+  {
+    tau = std::max(integrator_m.read_cell_data(penalty_parameters),
+                   integrator_p.read_cell_data(penalty_parameters)) *
+          IP::get_penalty_factor<dim, Number>(
+            degree,
+            GridUtilities::get_element_type(
+              integrator_m.get_matrix_free().get_dof_handler(dof_index).get_triangulation()),
+            data.IP_factor);
+  }
+
+  /**
+   * Recomputes `tau` before a boundary face evaluation.
+   */
+  void
+  reinit_boundary_face(IntegratorFace & integrator_m, unsigned int const dof_index) const
+  {
+    tau = integrator_m.read_cell_data(penalty_parameters) *
+          IP::get_penalty_factor<dim, Number>(
+            degree,
+            GridUtilities::get_element_type(
+              integrator_m.get_matrix_free().get_dof_handler(dof_index).get_triangulation()),
+            data.IP_factor);
+  }
+
+  /**
+   * Recomputes `tau` before a cell-based face evaluation.
+   */
+  void
+  reinit_face_cell_based(dealii::types::boundary_id const boundary_id,
+                         IntegratorFace &                 integrator_m,
+                         IntegratorFace &                 integrator_p,
+                         unsigned int const               dof_index) const
+  {
+    if(boundary_id == dealii::numbers::internal_face_boundary_id) // internal face
+    {
+      tau = std::max(integrator_m.read_cell_data(penalty_parameters),
+                     integrator_p.read_cell_data(penalty_parameters)) *
+            IP::get_penalty_factor<dim, Number>(
+              degree,
+              GridUtilities::get_element_type(
+                integrator_m.get_matrix_free().get_dof_handler(dof_index).get_triangulation()),
+              data.IP_factor);
+    }
+    else // boundary face
+    {
+      tau = integrator_m.read_cell_data(penalty_parameters) *
+            IP::get_penalty_factor<dim, Number>(
+              degree,
+              GridUtilities::get_element_type(
+                integrator_m.get_matrix_free().get_dof_handler(dof_index).get_triangulation()),
+              data.IP_factor);
+    }
+  }
+
+private:
+  //! Generalized Laplace kernel data
+  KernelData<dim> data{};
+
+  //! Element polynomial degree
+  unsigned int degree{1};
+
+  //! Internal penalty parameter
+  mutable scalar tau{0.0};
+
+  //! Geometry-dependent penalty parameters required for the calculation of `tau`
+  dealii::AlignedVector<scalar> penalty_parameters{};
+
+  //!
+  bool coefficient_is_variable{};
+
+  //! Constant coefficient
+  mutable coefficient_type constant_coefficient{};
+
+  //! Variable coefficients
+  mutable VariableCoefficients<coefficient_type> variable_coefficients{};
+};
+
+namespace Boundary
```

Comment:
This is superfluous and just leads to a duplication of names without further information, e.g. `Boundary::BoundaryType`, `Boundary::BoundaryDescriptor`.